spring-cloud-stream-binder-kstream/src/main/java/org/springframework/cloud/stream/binder/kstream/config/KStreamBinderConfiguration.java

```diff
@@ -0,0 +1,89 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.kstream.config;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.common.utils.AppInfoParser;
+import org.apache.kafka.streams.StreamsConfig;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.binder.kafka.admin.AdminUtilsOperation;
+import org.springframework.cloud.stream.binder.kafka.admin.Kafka11AdminUtilsOperation;
+import org.springframework.cloud.stream.binder.kafka.properties.KafkaBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner;
+import org.springframework.cloud.stream.binder.kstream.KStreamBinder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Condition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+/**
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ */
+@Configuration
+@EnableConfigurationProperties(KStreamExtendedBindingProperties.class)
+public class KStreamBinderConfiguration {
+
+	@Autowired(required = false)
+	private AdminUtilsOperation adminUtilsOperation;
+
+	private static final Log logger = LogFactory.getLog(KStreamBinderConfiguration.class);
+
+	@Bean
+	public KafkaTopicProvisioner provisioningProvider(KafkaBinderConfigurationProperties binderConfigurationProperties) {
+		return new KafkaTopicProvisioner(binderConfigurationProperties, adminUtilsOperation);
+	}
+
+	@Bean
+	public KStreamBinder kStreamBinder(KafkaBinderConfigurationProperties binderConfigurationProperties,
+			KafkaTopicProvisioner kafkaTopicProvisioner,
+			KStreamExtendedBindingProperties kStreamExtendedBindingProperties, StreamsConfig streamsConfig) {
+		return new KStreamBinder(binderConfigurationProperties, kafkaTopicProvisioner, kStreamExtendedBindingProperties,
+				streamsConfig);
+	}
+
+	@Bean(name = "adminUtilsOperation")
+	@Conditional(Kafka1xPresent.class)
+	@ConditionalOnClass(name = "kafka.admin.AdminUtils")
+	public AdminUtilsOperation kafka10AdminUtilsOperation() {
+		logger.info("AdminUtils selected: Kafka 0.10 AdminUtils");
```

Comment:
Isn't this going to confuse end-user since our real Kafka version is `0.11`?
Do we really need this info at all since we have one possible `AdminUtilsOperation` bean.
Plus bean name doesn't reflect reality - use use `Kafka11AdminUtilsOperation`